README.md

```diff
@@ -1,3 +1,149 @@
-# React Native Base
+# React Native Base - TypeScript
 
-This new base is under construction
+## Commands
+
+1. **Start the app**: `yarn android:{env}` or `yarn ios:{env}` (envs: `dev`, `staging`, and `prod`)
+2. **Start metro bundler**: `yarn start`
+3. **Lint the app**: `yarn lint`
+4. **Test the app**: `yarn test`
+
+## Getting started
+
+1. Install dependencies: `yarn`
+2. Create a `.env` file in the root directory of the project, based on the `.env.defaults` sample file and the extra constants that you may need. This will be your development env.
+
+You can also create `.env.prod` and `.env.staging` to define environment variables for production and staging.
+
+3. Rename your new project using `yarn rename`
+4. Start on android or ios: `yarn android:{env}` or `yarn ios:{env}` (envs: `dev`, `staging`, and `prod`)
+
+### Steps for Android development
+
+None
+
+### Steps for iOS development
+
+1. Run the following command inside the `ios` directory
+
+```
+pod install
+```
+
+## Continuous Integration
+
+The repo includes configuration for using GitHub Actions to run unit tests and code analysis: `.github/workflows/test.yml`. This can be adapted as needed for specifics of each project. Both CodeClimate and Sonarqube integrations are included in the workflow and their required environment settings should be retrieved from the repo Secrets.
+
+## Configuring Code Climate
+
+1. After adding the project to CC, go to `Repo Settings`
+2. On the `Test Coverage` tab, copy the `Test Reporter ID`
+3. Set the copied value as environment variable `CC_TEST_REPORTER_ID` (and repo Secrets)
+
+## Sonarqube Integration
+
+1. Log into Sonarqube console (`SONAR_URL`) and create new Project key(`SONAR_PROJECT`)
+2. Generate a token for the new project and copy
+3. Set the token value as environment variable `SONAR_TOKEN`
+
+### Usage
+
+```
+sonar-scanner \
+  -Dsonar.qualitygate.wait=true \
+  -Dsonar.host.url=$SONAR_URL \
+  -Dsonar.login=$SONAR_TOKEN \
+  -Dsonar.projectKey=$SONAR_PROJECT \
+  -Dsonar.scm.provider=git \
+  -Dsonar.java.binaries=/tmp \
+  -Dsonar.nodejs.executable=$(which node) \
+  -Dsonar.projectVersion=$(echo $GITHUB_SHA | cut -c1-8) \
+  -Dsonar.sources=. \
+  -Dsonar.projectBaseDir=. \
+  -Dsonar.javascript.lcov.reportPaths=coverage/lcov.info
+```
+
+## Build Android Release
+
+### Configuration
+
+🚧 UNDER CONSTRUCTION 🚧
+
+### Create release
+
+🚧 UNDER CONSTRUCTION 🚧
+
+## Build iOS Release
+
+1. Make sure that the version was already bumped if it applies.
+2. Select on Xcode the scheme of the env you want to create the release for.
+3. For the device select **generic iOS device**.
+4. Then go to **Product** -> **Archive**.
+5. After it is done processing and the archive succeeds the **organizer** will open. Here is where you can see all the previously generated archives.
```

Comment:
Maybe we can add a step 6 saying that you can upload to the App Store or as Ad Hoc? Otherwise someone may think that with the archive is already done.